File: Day-1/typescript/src/part2.ts
import * as fs from 'fs';

const data: [number, number][] = fs.readFileSync('../data.txt', 'utf8')
    .trim()
    .split('\n')
    .map(line => {
        const nums = line.split('   ').filter(s => s.length > 0).map(Number);
        return [nums[0], nums[1]] as [number, number];
    });
const data_first: number[] = data.map(a => a[0]).sort();
const data_second: number[] = data.map(a => a[1]).sort();
let sim_score: number = 0;
for (const x of data_first) {
    sim_score += x*data_second.filter(y => y === x).length;
}
console.log(sim_score);

File: Day-01/cpp/part2.cpp
#include <fstream>
#include <vector>
#include <iostream>
#include <algorithm>

int main() {
    std::ifstream file("../data.txt");
    std::vector<int> data_first, data_second;
    int x, y;
    while (file >> x >> y) {
        data_first.push_back(x);
        data_second.push_back(y);
    }
    int sim_score = 0;
    for(const int& x : data_first) {
        sim_score += x*std::count(data_second.begin(), data_second.end(), x);
    }
    std::cout << sim_score << std::endl;
}